### PR TITLE
Add support for personal access tokens request review API

### PR DESCRIPTION
**Pull request by joaopenteado**
Following up with my previous PR #2826, I am also proposing the following changes, which add support for [programmatically approving or denying](https://docs.github.com/en/rest/orgs/personal-access-tokens?apiVersion=2022-11-28#review-a-request-to-access-organization-resources-with-a-fine-grained-personal-access-token) fine-grained personal access token requests.

I thought about including the rest of this API, but I don't have an immediate need for it and the PR might get a little too big. I was also a bit unsure on the function signature, so let me know your thoughts.

### CODE REVIEW

File: github/orgs_personal_access_tokens.go
Comment (gmlewis) [inline github/orgs_personal_access_tokens.go:1]:
```suggestion
// Copyright 2023 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

File: github/orgs_personal_access_tokens_test.go
Comment (gmlewis) [inline github/orgs_personal_access_tokens_test.go:1]:
```suggestion
// Copyright 2023 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

File: github/orgs_personal_access_tokens.go
Comment (gmlewis) [inline github/orgs_personal_access_tokens.go:9]:
```suggestion
// ReviewPersonalAccessTokenRequest approves or denies a pending request to access organization resources via a fine-grained personal access token.
```

File: github/orgs_personal_access_tokens.go
Comment (gmlewis) [inline github/orgs_personal_access_tokens.go:16]: In this repo, we typically make structs like this be public options that are passed into the method, and one of the reasons for this is that GitHub has a history of adding new options to API endpoints. If we didn't do this, then we would be continually breaking the API of this client library by being forced to update the signature of this method to keep adding new fields.

So let's please follow the same pattern for this new endpoint as well.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @joaopenteado !

**Comment by joaopenteado** [comment]
@gmlewis thanks again for the quick feedback! I'll probably commit the requested changes by tomorrow.

> one of the reasons for this is that GitHub has a history of adding new options to API endpoints. If we didn't do this, then we would be continually breaking the API of this client library by being forced to update the signature of this method to keep adding new fields.

This makes perfect sense. I should've thought of that, thanks for the heads up!

Just two questions though, as I am still not very familiar with this package's approach to API design. I saw a lot of other methods using pointers to the `Options` struct containing the body payload. This is good in the sense that it could potentially avoid a copy of a big struct, but it also allows for passing in a `nil` pointer which could then likely be dereferenced later on and cause a `panic` or a runtime error in case a payload is required. I take it this is an accepted risk in this package's design?

Additionally, the `Options` structs themselves will often (_but strangely not always_) be composed of only reference types. I imagine this is done intentionally in order to distinguish between Go's default initialized values and the presence or absence of values in the option itself. However, this also could introduce a runtime panic if a required parameter is forgotten, is this also by design then?

In order words, despite `Action` being mandatory for this specify endpoint, I should design my `Options` struct like this:
```go
type ReviewPersonalAccessTokenRequestOptions struct {
	Action *string `json:"action,omitempty"`
	Reason *string `json:"reason,omitempty"`
}
```
And not like this:
```go
type ReviewPersonalAccessTokenRequestOptions struct {
	Action string `json:"action"`
	Reason *string `json:"reason,omitempty"` // or string
}
```

Is my understanding of the above points, correct?

**Comment by gmlewis** [comment]
> ... I saw a lot of other methods using pointers to the Options struct containing the body payload. ... I take it this is an accepted risk in this package's design?

So you will probably see inconsistencies in this regard within this client library, which I will take the blame for.
Ideally, if a method requires body parameters to be provided, then we would simply pass an `opts` parameter that is a struct passed by value (and not a struct pointer), and what I would recommend in this case.

> Additionally, the Options structs themselves will often (but strangely not always) be composed of only reference types. ...  is this also by design then?

Again, ideally, if a passed parameter is mandatory, we would not use a pointer, and we would not use `omitempty` in the JSON tag.  If a passed parameter is optional, then we use a pointer and include `omitempty`. There are also a few fiddly cases in the API where we must send a `null` or other weirdness where we take other measures, but I don't think this particular endpoint has any fiddly cases.

So it appears to me that your last suggestion would be preferred since `action` is mandatory and `reason` is not:

```go
type ReviewPersonalAccessTokenRequestOptions struct {
	Action string `json:"action"`
	Reason *string `json:"reason,omitempty"`
}
```

**Comment by gmlewis** [comment]
In case the last comment wasn't clear, I would think this method would have this in it:
`opts ReviewPersonalAccessTokenRequestOptions) ...`

(in other words, _not_ a pointer to a struct, but instead using pass-by-value.)

**Comment by joaopenteado** [comment]
@gmlewis Thank you for clarifying!
I've pushed some changes, let me know if you'd like me to change anything else.

**Comment by gmlewis** [comment]
> Do you think I should keep it as a `string` just in case GitHub changes their mind about this ID later, or switch it to a `int64`?

Since the official docs say that it is an "integer", then let's please make it an `int64` in the hopes that that will reduce developer confusion. Thanks.

**Comment by gmlewis** [comment]
Oh, and you can leave all the "%v" in the `fmt.Sprintf` as that is idiomatic Go and commonly used throughout this repo.  We only change this "%v" to other things when formatting must be non-standard (e.g. `"%05d"`).

**Comment by joaopenteado** [comment]
Done!

**Review by gmlewis — APPROVED** [review]
Thank you, @joaopenteado !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by liaodaniel — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @liaodaniel !
Merging.

**Pull request merged**
